doc/bytecode.md

```diff
@@ -307,23 +326,34 @@ Verification
 - For all instructions the opcode must be valid
 - Memory write instructions can use `r10` as destination register
 
+### until v1
+- No instruction can use `r11` as destination register
+
+### from v1
+- `add64 reg, imm` can use `r11` as destination register
+
 ### until v2
 - Opcodes from the product / quotient / remainder instruction class are forbiden
+- `neg32` and `neg64` are allowed
 - `le` is allowed
+- `lddw` (opcodes `0x18` and `0x00`) are allowed
 - `hor64` is forbidden
 - `callx` source register is encoded in the imm field
-- The targets of `call` instructions is checked at runtime not verification time
-- The offset of jump instructions must be limited to the range of the bytecode
 
 ### from v2
-- Every function must end in a `ja` or `exit` instruction
-- `lddw` (opcodes `0x18` and `0x00`) are forbidden
-- `neg32` and `neg64` are forbidden
 - Opcodes from the product / quotient / remainder instruction class are allowed
+- `neg32` and `neg64` are forbidden
 - `le` is forbidden
+- `lddw` (opcodes `0x18` and `0x00`) are forbidden
 - `hor64` is allowed
-- The offset of jump instructions must be limited to the range of the current function
 - `callx` source register is encoded in the src field
-- The targets of internal calls (`call` instructions with src ≠ 0) must have been registered at verification time
-- The targets of syscalls (`call` instructions with src = 0) must have been registered at verification time
-- `add64 reg, imm` can use `r11` as destination register
+
+### until v3
+- The targets of `call` instructions (which includes `syscall` instructions) is checked at runtime not verification time
+- The offset of jump instructions must be limited to the range of the bytecode
```

Comment:
Should this be the range of the function?

Reply:
This is the **before** v3 section.